articles/quickstart/spa/angular2/06-token-renewal.md

```diff
@@ -0,0 +1,169 @@
+---
+title: Silent Token Renewal
+description: This tutorial demonstrates how to add automatic access_token renewal to an Angular 2+ application with Auth0
+budicon: 448
+---
+
+<%= include('../../../_includes/_package', {
+  org: 'auth0-samples',
+  repo: 'auth0-angular-samples',
+  path: '06-Token-Renewal',
+  requirements: [
+    'Angular 2+'
+  ]
+}) %>
+
+## Server Setup
+
+Using access_token renewal will require a server component. A simplistic one could be done like this:
+
+```js
+const express = require('express');
+const app = express();
+const cors = require('cors');
+const staticFile = require('connect-static-file');
+
+app.use(cors());
+app.use('/silent', staticFile(`${__dirname}/silent.html`));
+
+app.listen(3001);
+console.log('Listening on http://localhost:3001');
+```
+
+You'll also need a file called `silent.html` in the same directory, and it should look like this:
+
+```html
+<!doctype html>
+<html>
+<head>
+  <meta charset="utf-8">
+  <script src="https://cdn.auth0.com/js/auth0/8.6/auth0.min.js"></script>
+  <script>
+    var AUTH0_CLIENT_ID = '${account.clientId}';
+    var AUTH0_DOMAIN = '${account.namespace}';
+
+    if (!AUTH0_CLIENT_ID || !AUTH0_DOMAIN) {
+      alert('Make sure to set the AUTH0_CLIENT_ID and AUTH0_DOMAIN variables in silent.html.');
+    }
+
+    var webAuth = new auth0.WebAuth({
+      domain: AUTH0_DOMAIN,
+      clientID: AUTH0_CLIENT_ID,
+      scope: 'openid profile',
+      responseType: 'token id_token',
+      redirectUri: 'http://localhost:4200'
+    });
+  </script>
+  <script>
+    webAuth.parseHash(window.location.hash, function (err, response) {
+      parent.postMessage(err || response, 'http://localhost:4200');
+    });
+  </script>
+</head>
+<body></body>
+</html>
+```
+
+This server will need to be running on port 3001 (it is hardcoded into the frontend). `silent.html` refers to `localhost:4200`, which is the port that `ng serve` will be serving the web app on.
+
+## Add Token Renewal
+
+At a basic level, to add token renewal, you'll need to add a function like this to your `AuthService`:
+
+```typescript
+public renewToken() {
+  this.auth0.renewAuth({
+    audience: AUTH_CONFIG.apiUrl,
+    redirectUri: 'http://localhost:3001/silent',
+    usePostMessage: true
+  }, (err, result) => {
+    if (err) {
+      alert(`Could not get a new token using silent authentication (${err.error}).`);
+    } else {
+      alert(`Successfully renewed auth!`);
+      this.setSession(result);
+    }
+  });
+}
+```
+
+This will load the silent callback page we added earlier in an invisible iframe, make a call to Auth0, and give us the result. However, this has to be called manually. You probably want it to be automated, so lets add some more functions:
```

Comment:
remove any "we" type words. "Lets" is another to remove. Use "you" instead. Here we can say

"The `renewToken` method will load the invisible iframe which is created in `silent.html`....